### PR TITLE
added fee_stats

### LINKED ISSUE #731: feat: add method to get current transaction fee
add `fee_stats` api to gosdk.

https://demo2.zus.network/miner01/v1/block/get/fee_stats

### PR DESCRIPTION
**Pull request by Hitenjain14**
### Changes
- Added `fee_stats` api to gosdk #731 .

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by cnlangzi — COMMENTED** [review]
I think we should query data from sharders, not miners?  @peterlimg  @dabasov  @guruhubb , is it right?

File: zcncore/transaction.go [hunk 1]
```diff
@@ -1096,6 +1096,44 @@ func GetChainStats(ctx context.Context) (b *block.ChainStats, err error) {
...
+
+	var numMiners = 4
+
+	if numMiners > len(_config.chain.Miners) {
+		numMiners = len(_config.chain.Miners)
+	}
+
+	var result = make(chan *util.GetResponse, numMiners)
+
+	queryFromMinersContext(ctx, numMiners, GET_FEE_STATS, result)
```
Comment (dabasov) [inline zcncore/transaction.go:1109]: please extend this func to work with miners and sharders
https://github.com/0chain/gosdk/blob/staging/zcncore/transaction_query.go#L316

**Review by dabasov — CHANGES_REQUESTED** [review]

File: zcncore/transaction.go (hunk 1 above)
Comment (Hitenjain14) [reply zcncore/transaction.go:1109]: You want to add `miners` to `TransactionQuery` struct and have similar function for miners ?

Comment (dabasov) [reply zcncore/transaction.go:1109]: we can pass func here
```go
host, err := tq.getRandomMiner(ctx)
```
https://github.com/0chain/gosdk/blob/b4f7de4f984909c9e4a0af40379ff365bb2bde3f/zcncore/transaction_query.go#L328

Comment (Hitenjain14) [reply zcncore/transaction.go:1109]: Whats the endpoint to check miners health ?

Comment (Hitenjain14) [reply zcncore/transaction.go:1109]: @dabasov I have extended the ForAny  to work for both miners and sharders , do u want the GetFeeStats  to use this or are we good with current implementation of it .

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | add-fee_stats  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**